egs/wsj/s5/steps/libs/tensorflow/kaldi_tensorflow/kaldiInterface.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2016    Vincent Renkens
+#                   Gaofeng Cheng
+##@package kaldiInterface
+#contains the functionality to interface with Kaldi
+
+import numpy as np
+import gzip
+from collections import OrderedDict
+import os
+from shutil import copyfile
+import subprocess
+from random import shuffle
+## shuffle the utterances and put them in feats_shuffled.scp
+#
+#@param featdir the directory containing the features in feats.scp
+def shuffle_examples(featdir):
+	#read feats.scp
+	featsfile = open(featdir + '/feats.scp', 'r')
+	feats = featsfile.readlines()
+	
+	#shuffle feats randomly
+	shuffle(feats)
+
+	#wite them to feats_shuffled.scp
+	feats_shuffledfile = open(featdir + '/feats_shuffled.scp', 'w')
+	feats_shuffledfile.writelines(feats)
+
+##read the alignment file generated by kaldi
+#
+#@param filename path to alignment file
+#
+#@return a dictionary containing the alignments with the utterance IDs as keys      
+def read_alignments(filename):
+	print("alifile_name: "+str(filename))
```

Comment:
this file seems to have an indent-level of 8, not 4 as we use for Kaldi code (and I think as required in google style). Make sure you are using spaces not tabs.  I mentioned in another PR what your emacs config should look like.... for vim setup, ask Vimal.